File: leetcode/0002-add-two-numbers/addTwoNumbers.test.ts
import cases from "./cases.json";
import addTwoNumbers from "./addTwoNumbers";
import { arrayToListNode } from "./helper";

test("add two numbers", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);

    expect(
      addTwoNumbers(
        arrayToListNode(testCase.input.l1),
        arrayToListNode(testCase.input.l2)
      )
    ).toStrictEqual(arrayToListNode(testCase.output));
  });
});

File: leetcode/0002-add-two-numbers/addTwoNumbers_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"reflect"
	"testing"

	"github.com/edram/leetcode/util"
)

type AddTwoNumbersCase struct {
	Input struct {
		L1 []int
		L2 []int
	}

	Output []int
}

func TestAddTwoNumbers(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]AddTwoNumbersCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("nums: %v ==> %v \n", testCase.Input, testCase.Output)

		result := addTwoNumbers(arrayToListNode(testCase.Input.L1), arrayToListNode(testCase.Input.L2))
		if !reflect.DeepEqual(*result, *arrayToListNode(testCase.Output)) {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
